chat-plugins/datatools.js

```diff
@@ -0,0 +1,968 @@
+									(dex[move].secondary.self.boosts[boost] < 1 && !searches[search][boost])) continue;
+							}
+							delete dex[move];
+						}
+					}
+					break;
+
+				case 'status':
+				case 'volatileStatus':
+					for (var searchStatus in searches[search]) {
+						for (var move in dex) {
+							if (dex[move][search] !== searchStatus) {
+								if (!dex[move].secondaries) {
+									if (!dex[move].secondary) {
+										if (searches[search][searchStatus]) delete dex[move];
+									} else {
+										if ((dex[move].secondary[search] !== searchStatus && searches[search][searchStatus]) ||
+											(dex[move].secondary[search] === searchStatus && !searches[search][searchStatus])) delete dex[move];
+									}
+								} else {
+									var hasSecondary = false;
+									for (var i = 0; i < dex[move].secondaries.length; i++) {
+										if (dex[move].secondaries[i][search] === searchStatus) hasSecondary = true;
+									}
+									if ((!hasSecondary && searches[search][searchStatus]) || (hasSecondary && !searches[search][searchStatus])) delete dex[move];
+								}
+							} else {
+								if (!searches[search][searchStatus]) delete dex[move];
+							}
+						}
+					}
+					break;
+
+				default:
+					return this.sendReplyBox("Something broke! PM SolarisFox here or on the Smogon forums with the command you tried.");
+			}
+		}
+
+		var results = [];
+		for (var move in dex) {
+			results.push(dex[move].name);
+		}
+
+		var resultsStr = targetMon ? ("<font color=#999999>Matching moves found in learnset for</font> " + targetMon + ":<br>") : "";
+		if (results.length > 0) {
+			if (showAll || results.length <= output + 5) {
+				results.sort();
+				resultsStr += results.join(", ");
+			} else {
+				resultsStr += results.slice(0, output).join(", ") + ", and " + string(results.length - output) + " more. <font color=#999999>Redo the search with 'all' as a search parameter to show all results.</font>";
+			}
+		} else {
+			resultsStr = "No moves found.";
+		}
+		return this.sendReplyBox(resultsStr);
+	},
+
+	learnset: 'learn',
+	learnall: 'learn',
+	learn5: 'learn',
+	g6learn: 'learn',
+	rbylearn: 'learn',
+	gsclearn: 'learn',
+	advlearn: 'learn',
+	dpplearn: 'learn',
+	bw2learn: 'learn',
+	learn: function (target, room, user, connection, cmd) {
+		if (!target) return this.parse('/help learn');
+
+		if (!this.canBroadcast()) return;
+
+		var lsetData = {set:{}};
+		var targets = target.split(',');
+		var template = Tools.getTemplate(targets[0]);
+		var move = {};
+		var problem;
+		var format = {rby:'gen1ou', gsc:'gen2ou', adv:'gen3oubeta', dpp:'gen4ou', bw2:'gen5ou'}[cmd.substring(0, 3)];
+		var all = (cmd === 'learnall');
+		if (cmd === 'learn5') lsetData.set.level = 5;
+		if (cmd === 'g6learn') lsetData.format = {noPokebank: true};
+
+		if (!template.exists) return this.sendReply("Pokemon '" + template.id + "' not found.");
+		if (targets.length < 2) return this.sendReply("You must specify at least one move.");
```

Comment:
There small discrepancy in line numbers is because I condensed  these two lines to remove the curly braces from the if statements. See what was previously lines 1115 - 1121 in the other file for comparison.